app/helpers/application_helper.rb

```diff
@@ -765,6 +765,26 @@ def _toolbar_chooser
     )
   end
 
+  # Calculate hash of toolbars to render
+  #
+  # keys are toolbar <div> names and values are toobar identifiers (now YAML files)
+  #
+  def calculate_toolbars
+    toolbars = {}
+    if inner_layout_present? # x_taskbar branch
+      toolbars['history_tb'] = history_toolbar_filename
+    elsif display_back_button? # taskbar branch
+      toolbars['summary_center_tb'] = controller.send(:restful?) ? "summary_center_restful_tb" : "summary_center_tb"
+    end
+
+    toolbars['center_tb'] = center_toolbar_filename
+    if fname = custom_toolbar_filename
+      toolbars['custom_tb'] = fname
+    end
+    toolbars['view_tb'] = x_view_toolbar_filename
```

Comment:
This is probably wrong... should be somehting like `toolbars['view_tb'] = inner_layout_present? ? x_view_toolbar_filename : view_toolbar_filename` ?